pkg/clients/tsclient/metric_definition.go

```diff
@@ -129,3 +129,39 @@ func GetLFM(def *Definition, labels []string) (string, error) {
 	}
 	return strings.Join(lfm, "\x00"), nil
 }
+
+// ParseMetricDelimited parses a delimited message
+func ParseMetricDelimited(s string) (map[string]string, error) {
+	x := strings.Split(s, "\x00")
+	if len(x)%2 != 1 {
+		return nil, fmt.Errorf("incomplete delimited metric")
+	}
+	labels := map[string]string{}
+	labels["__name__"] = x[0]
+	for i := 1; i < len(x); i += 2 {
+		key := x[i]
+		val := x[i+1]
+		labels[key] = val
+	}
+	return labels, nil
+}
+
+// ConvertLFMMapToPrometheusEncodedName converts a metric in the form map[string]string{"__name__":"sonar_cpu","__source__":"user", "cpu":"cpu0"}
+// to sonar_cpu{__source__="user",cpu="cpu0",host_id="899669",mode="iowait",user_id="208897"}
+func ConvertLFMMapToPrometheusEncodedName(lfm map[string]string) string {
+	name := lfm["__name__"]
+
+	labelStrings := make([]string, 0, len(lfm))
+
+	for label, value := range lfm {
+		if label == "__name__" {
+			continue
+		}
+		lbl := label + "=\"" + value + "\""
```

Comment:
use `fmt.Sprintf(`%s="%v"`, label, value)` so you're not escaping string delimiters

Reply:
good call

Reply:
fixed